conans/util/windows.py

```diff
@@ -102,3 +106,17 @@ def rm_conandir(path):
         short_path = load(link)
         rmdir(os.path.dirname(short_path))
     rmdir(path)
+
+
+def hashed_redirect(base, path, min_length=6, attempts=10):
+    max_length = min_length + attempts
+
+    full_hash = sha256(path.encode())
+    assert len(full_hash) > max_length
+
+    for length in range(min_length, max_length):
+        redirect = os.path.join(base, full_hash[:length])
```

Comment:
CI jobs should use their own short-paths cache, as they should use their own regular cache. 

Otherwise, the cache locks per package, are done per package, so for the cases where conan aims to provide some concurrency over the cache, I think it should be already covered.

Reply:
makes sense, thanks